python/bin/watchman-wait-aio

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env python3
+
+from __future__ import print_function
+
+from pywatchman.aioclient import AIOClient as WatchmanClient
+
+import argparse
+import asyncio
+import pywatchman
+import os
+import sys
+
+def fieldlist(s):
+    # helper for splitting a list of fields by comma
+    return s.split(',')
+
+parser = argparse.ArgumentParser(
```

Comment:
So, following your comment, I have heavily restructured the `watchman-wait` code to make it more modular.